[yuxunn] iP

yuxunn opened the pull request:
### Duke

> "Use Duke to plan your day! 😄  :D" - yuxunn

Duke is fuss free and easy to use! It is 
* text-based
* easy to use
* ~~hard to code out~~ VERY easy to use!

All you need to do is: 
1. download it from [here] (https://github.com/yuxunn/ip)
2. double-click it.
3. add your tasks
4. let it manage your tasks for you 🐢 

And it is **FREE!!!!!!**

Features: 
- [ ] Managing tasks
- [ ] Managing deadlines (coming soon!)
- [ ] Reminders (coming soon! ~~hopefully~~)

If you are a java programmer, you can use it to practice java too! Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
have fun with duke!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

yuxunn committed:
Level-0

yuxunn committed:
Level-2

yuxunn committed:
Level-2

yuxunn committed:
Level-3

yuxunn committed:
Level-4

yuxunn committed:
Level-5

yuxunn committed:
Level-6

yuxunn committed:
Level-6

yuxunn committed:
A-TextUiTesting

yuxunn committed:
Level-6

yuxunn committed:
Level-7

yuxunn committed:
Level-7

yuxunn committed:
Merge branch 'branch-Level-7'

yuxunn committed:
Level-8

yuxunn committed:
Merge branch 'branch-Level-8'

yuxunn committed:
A-MoreOOP

yuxunn committed:
A-Packages

yuxunn committed:
Merge branch 'branch-A-Packages'

yuxunn committed:
Merge branch 'add-gradle-support' into branch-A-Packages

yuxunn committed:
A-Gradle

yuxunn committed:
Merge branch 'branch-A-Gradle'

yuxunn committed:
A-JUnit

yuxunn committed:
Merge branch 'branch-A-JUnit'

yuxunn committed:
A-JavaDoc

yuxunn committed:
Level-9

euchangxian commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# Duke project template
+# duke.Duke project template
```
I am guessing the additional "duke." is unintentional. You could consider reverting this change.

euchangxian commented on README.md:
```diff
@@ -13,7 +13,7 @@ Prerequisites: JDK 11, update Intellij to the most recent version.
 1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
    In the same dialog, set the **Project language level** field to the `SDK default` option.
-3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
+3. After that, locate the `src/main/java/duke.Duke.java` file, right-click it, and choose `Run duke.Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
```
The file path should probably be `src/main/java/duke/Duke.java`.

euchangxian commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,65 @@
+//import java.io.IOException;
+//import java.io.FileReader;
+//import java.time.LocalDateTime;
+//import java.time.format.DateTimeFormatter;
```
Consider removing these unused imports to improve code cleanliness.

euchangxian commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+     * @return LocalDateTime of the parsed date and time.
+     */
+    private static LocalDateTime dateTask(String dateTimeStr) {
+        DateTimeFormatter DMYhelper = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
```
Variable names should be in camelCase as specified by [https://se-education.org/guides/conventions/java/intermediate.html](url). Consider renaming this variable to `dayMonthYearFormatter` or if too verbose, `dayMthYrFormatter`.

euchangxian commented on src/test/java/DukeTest.java:
```diff
@@ -0,0 +1,60 @@
+    public void testTodoCommand() {
+        Duke duke = new Duke("./data/duke.txt");
+
+        // Simulate user input to add a "todo" task
+        String simulatedUserInput = "todo Read a book\nbye\n";
+        InputStream inputStream = new ByteArrayInputStream(simulatedUserInput.getBytes());
+        System.setIn(inputStream);
+
+        // Redirect output to capture Duke's responses
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(outputStream));
```
Nice. I didn't know we could simulate user input like that. TIL.

euchangxian commented on text-ui-test/runtest.bat:
```diff
@@ -15,7 +15,7 @@ IF ERRORLEVEL 1 (
 
 REM run the program, feed commands from input.txt file and redirect the output to the ACTUAL.TXT
-java -classpath ..\bin Duke < input.txt > ACTUAL.TXT
+java -classpath ..\bin duke.Duke < input.txt > ACTUAL.TXT
```
This should be unintentional I believe. Consider reverting this!

jovkusuma commented on data/duke.txt:
```diff
@@ -0,0 +1,15 @@
+D | 0 | Finish report | null
+T | 0 | Read a book
+D | 0 | Finish report | null
+T | 0 | book
```
Shouldn't this file be ignored?

jovkusuma commented on data/tasks.txt:
```diff
@@ -0,0 +1,2 @@
+1
+D | 0 | return book | null
```
Why are there 2 separate files to store the tasks?

And shouldn't this file also be ignored?

jovkusuma commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,65 @@
+            String[] actions2 = new String[100];
+            boolean[] isDone2 = new boolean[100];
+            String[] types2 = new String[100];
+            taskList = new TaskList(actions2, isDone2, types2, 0);
```
Do not use numbers to distinguish names.

jovkusuma commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+            System.out.println("Bye. Hope to see you again soon!");
+            return true;
+        } else if (input.startsWith("delete")) {
+            int num3 = Integer.parseInt(input.substring(7).trim());
```
I noticed the same issue in several other places too..

jovkusuma commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+                TaskList.isDone[j] = TaskList.isDone[j + 1];
+            }
+            TaskList.counter = TaskList.counter - 1;
+            System.out.println("Noted. I've removed this task:\n" + action2);
```
I noticed the same issue in several other places too..

jovkusuma commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+                return false;
+            }
+        } else if (input.startsWith("unmark")) {
+            int num2 = Integer.parseInt(input.substring(7).trim());
```
I noticed the same issue in several other places too..

jovkusuma commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+        isDone[counter] = false;
+        actions[counter] = action;
+        type[counter] = "T";
+        counter= counter + 1;
```
Shouldn't there be a space before "="
```suggestion
        counter = counter + 1;
```

jovkusuma commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+     * @return Formatted LocalDateTime version.
+     */
+    private static LocalDateTime helper2 (String dateTimeStr) {
+        DateTimeFormatter DMYhelper = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
```
Shouldn't this be in camelCase?

```suggestion
        DateTimeFormatter dmyHelper = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
```

jovkusuma commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,161 @@
+    }
+
+    /**
+     * Saves the task data to a file for future retrival.
+     * @param fileName name of file to save data.
```
Shouldn't there be a space between the description and the parameter section

jovkusuma commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,161 @@
+    }
+
+    /**
+     * Loads an array whether the activity is done.
+     * @return Array of whether the task is done.
```
Shouldn't there be a space between the description and the return section?

```suggestion
     * Loads an array whether the activity is done.
     *
     * @return Array of whether the task is done.
```

jovkusuma commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,161 @@
+    }
+
+    /**
+     * An array of type of activity.
+     * @return Array of activity's type.
```
I noticed the same issue in several other places too..

jovkusuma commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,161 @@
+    }
+
+    /**
+     * Loads the array of action.
+     * @return An array of actions.
```
I noticed the same issue in several other places too..

jovkusuma commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+
+
+    /**
+     * Constructor of the TaskList
+     * @param actions Array of actions.
+     * @param isDone Array of whether the action is done.
+     * @param type Array of the different type of actions.
+     * @param counter Number of action in tasklist.
```
I noticed the same issue in several other places too..

jovkusuma commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    /**
+     * Formats and saves the to do type of action.
+     * @param action String of name of action.
```
I noticed the same issue in several other places too..

jovkusuma commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+
+
+    /**
+     * Helper method to format the type of action.
+     * @param task Name of task.
+     * @param taskType Type of task.
+     * @param isDone True if done, false if not done.
+     * @return String with formatted type of tasktype.
+     */
```
I noticed the same issue in several other places too..

jovkusuma commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    /**
+     * Helper method to format and save the deadline type of task.
+     * @param action Name of task.
+     * @param due String of when the task is due.
+     * @param by LocalDateTime of formatted due.
+     */
```
I noticed the same issue in several other places too..

jovkusuma commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+     * @param action Name of action.
+     * @param start String indicating start time.
+     * @param end String indicating the end time.
+     */
```
I noticed the same issue in several other places too..

jovkusuma commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    /**
+     * Helper method to format the string into the LocalDateTime format.
+     * @param dateTimeStr String of the to-be-formatted date time.
+     * @return Formatted LocalDateTime version.
```
I noticed the same issue in several other places too..

jovkusuma reviewed (commented):
You can take a look at my comments to see which ones to fix regarding naming as well as coding standard, besides that, not a bad job!

euchangxian reviewed (commented):
Overall LGTM, with some minor style issues here and there, but those are easy fixes! Good job!

yuxunn committed:
branch-Level-10

yuxunn committed:
Merge branch 'branch-Level-10'

yuxunn committed:
branch-A-Assertions

yuxunn committed:
Improve code quality

yuxunn committed:
Merge pull request #1 from yuxunn/branch-A-CodeQuality

Improve code quality

yuxunn committed:
C-Help

yuxunn committed:
Add Help Functionality.

This commit adds a new feature to the Duke Application.

Users can now access a HELP page by typing help as their input.

Within Help page contains list of commands users can use in Duke.

This functionality will improve user experience by providing them with the knowledge on how to use the different features of the application.

yuxunn committed:
Add assertions to application to improve code quality.

This commit uses assertions to document important assumptions in the Duke application implementation.

Changes made:
 - Added assert statements to find method, inputType method, deadline method.
 - Improved error detection.

This commit improves reliability of the application.

yuxunn committed:
Merge pull request #3 from yuxunn/branch-A-Assertions

Add assertions to application to improve code quality.

yuxunn committed:
Update README.md

yuxunn committed:
Update README.md

yuxunn committed:
Add files via upload

yuxunn committed:
Update for JAR

yuxunn committed:
Merge branch 'branch-V2.0'

yuxunn committed:
Merge remote-tracking branch 'origin/master'
